tests/ui/needless_return.stderr

```diff
@@ -133,12 +133,10 @@ LL +     the_answer!()
    |
 
 error: unneeded `return` statement
-  --> tests/ui/needless_return.rs:86:21
+  --> tests/ui/needless_return.rs:87:5
    |
-LL |   fn test_void_fun() {
-   |  _____________________^
-LL | |     return;
```

Comment:
I'm not entirely convinced that we shouldn't extend this span upto at least the full line?

Reply:
Maybe extend the span to the `;`. Not sure why it wasn't included in the previous span though. There might've been a reason.

Might need some logic like "if parent statement is a `StmtKind::Semi`, extend the span to the statement" :thinking:

Reply:
The semicolon spans are handled separately in the suggestion since there may be multiple of them.